### PR TITLE
Blog app: add devise

### PR DESCRIPTION
**Pull request by RayhanTabase**
In this feature:
- Removed current_user method in ApplicationController
- Install and setup devise
- Can register a new user.
- User logs in with a combination of email and password.
- Hashed passwords stored in the database.
- Ask for confirmation of email.
- Can reset password.
- Modified the views in devise to match wireframe and styling

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -1,6 +1,11 @@
 class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
```
Comment (Gambit142) [inline app/models/user.rb]: - [ ] One of the project's requirements is that users should confirm their email before they can navigate the application. Kindly add `:confirmable` to the attributes and implement the feature for confirming email. 👍

**Review by ignatius22 — CHANGES_REQUESTED** [review]
Hi @RayhanTabase, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights


## Required Changes ♻️
_Check the comments under the review._
- [ ] I think it would be nice to include CanCan Gem in Gemfile: gem 'cancancan', since it's part of the project requirement.
- [ ] `role` column is missing in the users table - kindly update migration files for this.
- [ ] Let's try and follow the project requirements by including CanCanCan `ability` class that defines the set of rules for user-post interactions.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Gambit142 — CHANGES_REQUESTED** [review]
Hi @RayhanTabase @direakanbi 👋, 

Good job so far!

## Highlights
- Removed `current_user` method in `ApplicationController` ✔️ 
- Installed and set up `Devise` correctly ✔️ 
- User can log in with a combination of `email` and `password` ✔️ 

However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Kindly check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider

N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Review by Meltrust — APPROVED** [review]
Hi @RayhanTabase,


## Wow, you did it 🎉
Brilliant
## Thank you for implementing the changes requested by the previous reviewer 💪 🥇 ㊗️ 
### Well done!




Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

### To highlight

- Thank you for adding the devise confirmable module, as required✔️ 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**